File: app/src/androidTest/java/com/handstandsam/shoppingapp/features/login/ErrorScenariosTest.java
package com.handstandsam.shoppingapp.features.login;


import android.support.test.rule.ActivityTestRule;

import com.handstandsam.shoppingapp.R;
import com.handstandsam.shoppingapp.features.home.HomeActivity;

import org.junit.Rule;
import org.junit.Test;

import static android.support.test.espresso.Espresso.onView;
import static android.support.test.espresso.action.ViewActions.click;
import static android.support.test.espresso.assertion.ViewAssertions.matches;
import static android.support.test.espresso.contrib.RecyclerViewActions.actionOnItemAtPosition;
import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
import static android.support.test.espresso.matcher.ViewMatchers.withId;
import static android.support.test.espresso.matcher.ViewMatchers.withText;
import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
import static com.github.tomakehurst.wiremock.client.WireMock.get;
import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
import static org.hamcrest.Matchers.allOf;

public class ErrorScenariosTest {

    @Rule
    public ActivityTestRule<HomeActivity> mActivityTestRule
            = new ActivityTestRule<>(HomeActivity.class);

    @Test
    public void http500ErrorTest() {
        stubFor(get(urlEqualTo("/category/Nintendo/items"))
                .willReturn(aResponse().withStatus(500)));

        onView(allOf(withId(R.id.categories), isDisplayed()))
                .perform(actionOnItemAtPosition(0, click()));

        onView(withText("Networking Error")).check(matches(isDisplayed()));
    }
}

File: app/src/androidTest/java/com/handstandsam/shoppingapp/features/login/ErrorScenariosTest.kt
package com.handstandsam.shoppingapp.features.login


import android.content.Intent
import androidx.test.InstrumentationRegistry
import androidx.test.rule.ActivityTestRule
import androidx.recyclerview.widget.RecyclerView
import android.view.View
import androidx.test.espresso.Espresso.onView
import androidx.test.espresso.action.ViewActions.click
import androidx.test.espresso.assertion.ViewAssertions.matches
import androidx.test.espresso.contrib.RecyclerViewActions.actionOnItemAtPosition
import androidx.test.espresso.matcher.ViewMatchers.*
import com.handstandsam.shoppingapp.R
import com.handstandsam.shoppingapp.features.home.HomeActivity
import com.handstandsam.shoppingapp.models.User
import com.handstandsam.shoppingapp.preferences.UserPreferences
import org.hamcrest.Matchers.allOf
import org.junit.Before
import org.junit.Rule
import org.junit.Test

class ErrorScenariosTest {

    @Rule
    @JvmField
    var mActivityTestRule = ActivityTestRule(HomeActivity::class.java, true, false)

    @Before
    fun setUp() {
        val context = InstrumentationRegistry.getTargetContext()
        val userPreferences = UserPreferences(context)
        userPreferences.currentUser= User("Sam", "Edwards")
        mActivityTestRule.launchActivity(Intent(context, HomeActivity::class.java))
    }

    @Test
    fun http500ErrorTest() {
//        stubFor(get(urlEqualTo("/category/Nintendo/items"))
//                .willReturn(aResponse().withStatus(500)))

        onView(allOf(withId(R.id.categories), isDisplayed()))
                .perform(actionOnItemAtPosition<RecyclerView.ViewHolder>(0, click()))

        onView(withText("Networking Error")).check(matches(isDisplayed()))
    }
}
